### PR TITLE
Add specs

### PR DESCRIPTION
**Pull request by pjambet**
I added 2 small specs, just to make sure that the new signal handling logic worked as expected. I'm not a fan of how I had to write the specs (using a separate thread to run the worker), but it works, which was my initial goal!

I also refactored logging a tiny bit, to make sure it was consistent with how Resque does it and so it can be controller through the verbosity env variables.

PS: Thank you so much for writing this gem 👍

### CODE REVIEW

File: lib/resque/heroku-signals.rb [hunk 1]
```diff
@@ -6,16 +6,16 @@
   def unregister_signal_handlers
     trap('TERM') do
       trap('TERM') do
-        puts "[resque-heroku] received second term signal, throwing term exception"
+        log_with_severity :info, "[resque-heroku] received second term signal, throwing term exception"
 
         trap('TERM') do
-          puts "[resque-heroku] third or more time receiving TERM, ignoring"
+          log_with_severity :info, "[resque-heroku] [resque-heroku] third or more time receiving TERM, ignoring"
```
Comment (iloveitaly) [inline lib/resque/heroku-signals.rb]: @pjambet looks like there's a duplicate `[resque-heroku]` here. Could you remove this?

**Comment by iloveitaly** [comment]
@pjambet awesome improvements! Thanks so much. I requested one quick change, then I'll merge it in.

File: lib/resque/heroku-signals.rb (hunk 1 above)
Comment (pjambet) [reply lib/resque/heroku-signals.rb]: Oops! Thanks for catching this

**Comment by pjambet** [comment]
Thanks for the quick review, I fixed it and rebased, should be ready now.

**Pull request merged**